lib/cc/config/yaml_adapter.rb

```diff
@@ -55,6 +55,9 @@ def upconvert_legacy_yaml!
           plugins = config.fetch("plugins", {})
           plugins.each do |engine, data|
             plugins[engine] = coerce_engine(data)
+            if plugins.fetch(engine)["exclude_paths"]
+              plugins.fetch(engine)["exclude_patterns"] ||= plugins.fetch(engine).delete("exclude_paths")
```

Comment:
nice use of `||=` to express that it shouldn't override the `exclude_patterns` when present. Worth covering that precedence behavior with a test?

Reply:
Test added, will merge on green.